src/routes/Home.jsx

```diff
@@ -1,8 +1,17 @@
 import useCurrentUserData from '../utils/useCurrentUserData';
-
+import SignInButton from '../SignInButton';
+import SignOutButton from '../SignOutButton';
+import { onAuthStateChanged } from 'firebase/auth';
+import { auth } from '../utils/firebase';
 const Home = () => {
   const { userData, setUserData } = useCurrentUserData();
-
+  onAuthStateChanged(auth, (user) => {
+    if (user) {
+      setUserData({ username: user.displayName, email: user.email, uid: user.uid });
```

Comment:
let's match the key name as the same as in firebase. userid instead of 106292803